tests/dialects/test_duckdb.py

```diff
@@ -339,6 +345,12 @@ def test_duckdb(self):
                 "bigquery": "ARRAY_CONCAT([1, 2], [3, 4])",
             },
         )
+        self.validate_all(
+            "SELECT CAST(CAST(x AS DATE) AS DATE) + INTERVAL 1 DAY",
```

Comment:
the double cast is awkward but there's another test that expects the explicit cast. seems more hacky to try to inspect the rendered sql for 'this' to see if we need to cast again. open to other suggestions...

Reply:
checking if `isinstance(this, exp.Cast) and this.is_type(exp.DataType.Type.DATE)` doesn't seem _too_ bad to me.. but I can go either way. 

This is easier with type annotations, but alas, that only happens during optimization.

Reply:
it's not that straightforward tho bc we don't know if it's a Cast until we render it at which point we'd have to reparse

Reply:
(but that might be overkill)

Reply:
this is ok, this optimaziton should go in simplify